template/registry_number.go

```diff
@@ -22,23 +22,24 @@ func parseNumberOperand(operand any) (float64, error) {
 		return 0, fmt.Errorf(format+": %w", append(args, mf2.ErrBadOperand)...)
 	}
 
-	if operand == nil {
-		return errorf("operand is required")
-	}
-
 	var (
 		v   float64
 		err error
 	)
 
-	if s, ok := operand.(string); ok {
-		err = json.Unmarshal([]byte(s), &v)
-	} else {
+	switch t := operand.(type) {
```

Comment:
```suggestion
        var (
            number float64
        )
	switch v := operand.(type) {
```

Reply:
How is `t` and `v` better?

`v` is idiomatic in Golang to represent a value, and in the given scenario, its scope is limited to the switch case, and it represents a typed value. Easy to follow its consumption.

`number` says it all. It is a number.